storage/store.go

```diff
@@ -1156,6 +1154,7 @@ func (s *Store) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.Bat
 	for _, union := range ba.Requests {
 		arg := union.GetInner()
 		header := arg.Header()
+		// TODO(tschottdorf): something fishy here
```

Comment:
removed (not sure what I thought was fishy - some of these checks could operate on `keys.Range(ba)` instead, but that's a) performance b) gives worse errors).